### ISSUE TITLE
Tests call model methods which are only implemented in backend

### ISSUE THREAD

**Issue by russelljjarvis** [scidash/neuronunit#112]
every model call to inject_square_current in tests/__init__.py (there is rather a lot), needs to be called with `model._backend.inject_square_current`. Unless I fix models/__init__.py to have its own 
```python 
def  self.inject_square_current(self,current):
   self._backend.inject_square_current(current)
```` 
method, which I will do tomorrow.

**Comment by rgerkin** [comment]
@russelljjarvis 
This is going to occur in more places than I anticipated.  Two fixes that I think will be needed:  
- Change the LEMSModel so that `__getattr__` first checks its own attributes and then checks the backend's attributes.  Then it will just automatically find whichever one exists.  I implemented this in d2a00ffc9581a3f63d2c6283ee07178e16b541eb.  This was the "level 4" solution I talked about the other day.  
- Those backend methods have a `self` which refers to the backend and not the model.  Methods written into the backend may need to explicitly call `self.model` rather than `self` whenever they are accessing attributes or methods of the model and not of the same backend.  You could try just doing this when it comes up.  

I have no idea if this is the complete solution, or this will also lead to a bunch of problems.  I feel like it can't be worse than the previous solution of dynamically editing the class bases.

**Comment by rgerkin** [comment]
@russelljjarvis 
I fixed enough of these to pass my unit tests, but there are probably still more to fix especially in the `NEURONBackend`.  My principle for deciding how to fix them is as follows: if the attribute X is something that ought to exist independently of the backend (e.g. attributes of the model), and the backend needs to access it, then the backend should access it at `self.model.X`.  If it is something that really belongs only to that specific backend, like locations of hoc files, then it can just be `self.X` (from inside a backend method).  Conversely, if a model method needs to access a backend attribute it can do so at self._backend.X, but this is less common.  

Also, I ended up scrapping the __getattr__ override because I foresaw some problems that might arise.  Instead, I just directly added passthrough methods as needed to the model.  For example, if a test wants the model to call method X, then I just added a wrapper method X to LEMSModel which contains only `return self._backend.X(*args,**kwargs)`.  All it does is just wrap the backend method so the test never has to actually access the backend attribute.  The test shouldn't ever directly call the backend methods.  You will see some examples of what I am talking about in commits I made today that added wrapper methods to the `LEMSModel` class.
